crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -633,58 +653,128 @@ impl IndexerReader {
                 let object_id = Hex::encode(object_id.to_vec());
-                format!("input_objects @> ARRAY['\\x{}'::bytea]", object_id)
+                (
+                    "tx_input_objects".into(),
+                    format!("object_id = '\\x{}'::bytea", object_id),
+                )
             }
             Some(TransactionFilter::ChangedObject(object_id)) => {
                 let object_id = Hex::encode(object_id.to_vec());
-                format!("changed_objects @> ARRAY['\\x{}'::bytea]", object_id)
+                (
+                    "tx_changed_objects".into(),
+                    format!("object_id = '\\x{}'::bytea", object_id),
+                )
             }
             Some(TransactionFilter::FromAddress(from_address)) => {
                 let from_address = Hex::encode(from_address.to_vec());
-                format!("senders @> ARRAY['\\x{}'::bytea]", from_address)
+                (
+                    "tx_senders".into(),
+                    format!("sender = '\\x{}'::bytea", from_address),
+                )
             }
             Some(TransactionFilter::ToAddress(to_address)) => {
                 let to_address = Hex::encode(to_address.to_vec());
-                format!("recipients @> ARRAY['\\x{}'::bytea]", to_address)
+                (
+                    "tx_recipients".into(),
+                    format!("recipient = '\\x{}'::bytea", to_address),
+                )
             }
             Some(TransactionFilter::FromAndToAddress { from, to }) => {
                 let from_address = Hex::encode(from.to_vec());
                 let to_address = Hex::encode(to.to_vec());
-                format!(
-                    "(senders @> ARRAY['\\x{}'::bytea] AND recipients @> ARRAY['\\x{}'::bytea])",
-                    from_address, to_address
-                )
+                // Need to de-amibguize the tx_sequence_number column
+                let cursor_clause = if let Some(cursor_tx_seq) = cursor_tx_seq {
+                    if is_descending {
+                        format!(
+                            "AND tx_senders.{TX_SEQUENCE_NUMBER_STR} < {}",
+                            cursor_tx_seq
+                        )
+                    } else {
+                        format!(
+                            "AND tx_senders.{TX_SEQUENCE_NUMBER_STR} > {}",
+                            cursor_tx_seq
+                        )
+                    }
+                } else {
+                    "".to_string()
+                };
+                let inner_query = format!(
+                    "(SELECT tx_senders.{TX_SEQUENCE_NUMBER_STR} \
+                    FROM tx_senders \
+                    JOIN tx_recipients \
+                    ON tx_senders.{TX_SEQUENCE_NUMBER_STR} = tx_recipients.{TX_SEQUENCE_NUMBER_STR} \
+                    WHERE tx_senders.sender = '\\x{}'::BYTEA \
+                    AND tx_recipients.recipient = '\\x{}'::BYTEA \
+                    {} \
```

Comment:
It's fine here because this order by is after the projection, namely it orders the final results which only has one column called `tx_sequence_number`